src/prefect_server/api/flows.py

```diff
@@ -579,19 +579,31 @@ async def schedule_flow_runs(flow_id: str, max_runs: int = None) -> List[str]:
     # schedule every event with an idempotent flow run
     for event in flow_schedule.next(n=max_runs, return_events=True):
 
+        # TODO: This has potential to miss scheduling runs that all need to be scheduled at the same time
```

Comment:
Can we link this todo to a github issue?

Reply:
Yeah I can open a followup issue 👍

Reply:
I don't have an issue yet to link it to, this is more of a visual edge case from looking at the code where I can't prove it happening (and don't have a reproducible example) but the assumption holds true

Reply:
@joshmeek I believe this should do the trick:
```python
clocks = [
    prefect.schedules.clocks.IntervalClock(
        start_date=now.add(minutes=1),
        interval=datetime.timedelta(minutes=2),
        parameter_defaults=dict(x=i),
    )
    for i in range(20)
]

flow = prefect.Flow(
    tasks=[Parameter("x", default=None)],
    name="Test Scheduled Flow",
    schedule=prefect.schedules.Schedule(clocks=clocks),
)
flow_id = await api.flows.create_flow(
    project_id=project_id, serialized_flow=flow.serialize()
)
await models.FlowRun.where({"flow_id": {"_eq": flow_id}}).delete()
assert len(set((await api.flows.schedule_flow_runs(flow_id)))) == 10
assert len(set((await api.flows.schedule_flow_runs(flow_id)))) == 10 # this will fail in your implementation
```

Reply:
errr sorry that might not be correct, it might still schedule 10 runs just not the correct runs - you'll need to test for all 20 runs with the correct parameter values

Reply:
@cicdw A bit confused, are you saying we should make a test where we have the expectation that it will fail since it will in the current implementation?

Reply:
No, I was suggesting a reproducible example for an issue